ManagePeriods to Production

vlagrassa opened the pull request:
Update Production to match recent work in Vue branch, mostly to do with `ManagePeriods` and other period/class pieces.

**Major/Relevant Updates**
- Both `ManagePeriods` and Youth Profile pages now read youth period/class data from `GlobalPeriods`, meaning the youth's `ActivePeriods` field is now obsolete.
- Moved Profile popup modal to its own component, which can be used on any page as long as the ID is given.  If you already have header/profile information loaded, it's possible to pass it in through props (to cut down on reads & for customizability), but it can also retrieve that on its own if none is given.
- `AwaitTransactions` object in `ParseDB.js` will take a series of Firebase transactions, perform them all, and run a given success/failure callback after all transactions come back. Makes it easier to manage/synchronize updates to multiple locations in the database.
- Use `ModalDRS` component to handle edits (Discard, Reset, and Save buttons linked to a series of popup modals).

**Miscellaneous Updates**
- Item logs in profile pages should now initially sort by date (originally sorted by document ID on Firebase)
- Hour display divs on profile page now have their own component
- New `mapObj` function in `ParseDB.js`
- Warning badges on youth fields only display in staff view, and can be (de)activated with a prop

vlagrassa committed:
Sort profile log tables by date initially

See cb96e3e69439b68bbbe1151ff3a9cbd3557b0a9a

vlagrassa committed:
Create mapObj helper function

See 7fd44f75ca058cfdfad5e1ae48c83cd4d6a1097b

vlagrassa committed:
Repurpose SaveBar as ModalDRS (delete reset save)

See c5c407ed2360c17eb2602909ca92943f084823c6

vlagrassa committed:
Ensure concat_overwrite makes and returns changes in place

See cf6dc40a99bd648adaaedcd61148c59e3a39fdc3

vlagrassa committed:
Move profile popup modal to new component

See b40c08db7494708ddae293e59fe2cd165a858ce6

vlagrassa committed:
Remove unused Array type

See 8de27e850596404161e4446a14eab47a5582b1b2

vlagrassa committed:
Make component for hour decimal displays

See ab5ecf5d30457a46f9ea1244db42dbadbce8aec5

vlagrassa committed:
Use ModalDRS instead of SaveBar

See c5c407ed2360c17eb2602909ca92943f084823c6 and 6d46b6c13c43ef3e197a3cc974a459d69704b5ee

vlagrassa committed:
Actually compute youth name and ID

vlagrassa committed:
Use AwaitTransactions object to update database

See b0a254cf0a2a6604e9f1a47b246eb17b76590182 and 86ce2ac3fccb82fbac4e6d70b088e5788924cd36

vlagrassa committed:
Split up imports to look nicer

vlagrassa committed:
Move current quarter header inside buttons

vlagrassa committed:
Update disableWarnings prop

vlagrassa committed:
Add hideTitle prop for popup

See b40c08db7494708ddae293e59fe2cd165a858ce6

vlagrassa committed:
Move Active Periods Display out of ProfileFields

See a19fee1268fae6a5a0ba050295f057b85c9a7a4a and 8c92b7ecd8b0048e1122f1d8f23c4fbf4b520513

vlagrassa committed:
Disable field warnings in youth view

See f4fbf15953e2d065e28003e0ae000d69bd2adaa5

vlagrassa committed:
Minor cleanup

vlagrassa committed:
Keep the changed CSS class

yvesshum commented:
Given current period = Spring, registration = Summer, ManagePeriods.js  returns an error when trying to save a new summer class. Editing the current period's classes and before works fine though

katedlu commented:
After successfully submitting changes, the chosen changes aren't cleared from the component

vlagrassa committed:
Allow custom element in ModalDRS failure footer

vlagrassa committed:
Only let save fail modal close on OK button

vlagrassa committed:
Show failed updates properly

vlagrassa committed:
Create new period array if it doesn't exist yet

Fixes #65

vlagrassa committed:
Draw log hour categories from database

vlagrassa committed:
Show default cursor on future periods

vlagrassa committed:
Keep track of proper old_class, remove change if manually reset

Fixes #68

vlagrassa committed:
Force YouthIDSelectors to open downward in profile lookup pages

vlagrassa committed:
Change current period button to home icon

vlagrassa committed:
Overwrite local values after database update

Fixes #64

vlagrassa committed:
Wait to show button header title until display period set

Fixes #48

vlagrassa committed:
Separate class and class string functions

Used to replace null with "Not Active" when setting a non-active period
to non-active

vlagrassa committed:
Make UI friendlier

Fixes #67

vlagrassa committed:
Use YouthIDSelector to choose youth from any period

vlagrassa committed:
Display more helpful message

vlagrassa committed:
Put Profile and Discard/Save buttons in line with one another

vlagrassa committed:
Allow YouthIDSelector to hide certain youth with prop

vlagrassa committed:
Change name to proper 'profile-popup'

vlagrassa committed:
Remove spacing

vlagrassa committed:
Change amount of space for text vs button

vlagrassa committed:
Better warning message

vlagrassa committed:
Set default prop array with anonymous function

vlagrassa committed:
Show warning badge when adding class for registration period

katedlu reviewed (approved):
Looks great!

yvesshum committed:
Merge branch 'production' of https://github.com/yvesshum/bikeshop-webapp into ManagePeriodsToProduction

yvesshum reviewed (approved):
Nice work!
